### ISSUE TITLE
Mylar won't start on Ubuntu 18.04

### ISSUE THREAD

**Issue by ljw1** [evilhero/mylar#1937]
Newly upgraded box running Ubuntu 18.04 won't start mylar. Checked bs4 and html5lib and both are up to date.

commit 78d84eee5ed36589ec91e7f742383161e35f9f42 (HEAD -> development, origin/development)
Author: evilhero <evilhero@gmail.com>
Date:   Fri Apr 27 21:23:04 2018 -0400

python /opt/mylar/Mylar.py -d
Traceback (most recent call last):
  File "/opt/mylar/Mylar.py", line 26, in <module>
    import mylar
  File "/opt/mylar/mylar/__init__.py", line 41, in <module>
    from mylar import logger, versioncheckit, rsscheckit, searchit, weeklypullit, PostProcessor, updater, helpers
  File "/opt/mylar/mylar/rsscheckit.py", line 21, in <module>
    from mylar import logger, rsscheck, helpers, auth32p
  File "/opt/mylar/mylar/rsscheck.py", line 17, in <module>
    from mylar import db, logger, ftpsshup, helpers, auth32p, utorrent
  File "/opt/mylar/mylar/auth32p.py", line 9, in <module>
    from bs4 import BeautifulSoup
  File "/opt/mylar/bs4/__init__.py", line 29, in <module>
    
  File "/opt/mylar/bs4/builder/__init__.py", line 306, in <module>
    
  File "/opt/mylar/bs4/builder/_html5lib.py", line 57, in <module>
AttributeError: 'module' object has no attribute '_base'

**Comment by evilhero** [comment]
I think that the bs4 included with mylar has to be updated based on some readings. Seems there's a bug if you're up to date on html5lib (which would be loaded from your system) and using a not up to date bs4 (which is what mylar uses).

Can you, with mylar not running, go into the lib directory off of the root of mylar and just rename the bs4 directory to something else (ie. bs4.backup). If my hunch is right, mylar will then be forced to load bs4 off your system instead of using the lib version and it will load since both html5lib & beautifulsoup are current.

**Comment by ljw1** [comment]
Moving the bs4 folder and starting up works perfectly.

The versions that came with the system version of ubuntu 18.04 had updates available, which is what I tried before submitting the bug request.

pip show html5lib bs4
Name: html5lib
Version: 1.0.1
Summary: HTML parser based on the WHATWG HTML specification
Home-page: https://github.com/html5lib/html5lib-python
Author: James Graham
Author-email: james@hoppipolla.co.uk
License: MIT License
Location: /usr/local/lib/python2.7/dist-packages
Requires: six, webencodings

pip show beautifulsoup4
Name: beautifulsoup4
Version: 4.6.0
Summary: Screen-scraping library
Home-page: http://www.crummy.com/software/BeautifulSoup/bs4/
Author: Leonard Richardson
Author-email: leonardr@segfault.org
License: MIT
Location: /usr/lib/python2.7/dist-packages
Requires:

**Comment by Serneum** [comment]
Just a heads-up that I also hit this in the past week after upgrading to 18.04. Renaming `bs4` in `lib` didn't help as Mylar was using a `bs4` from its root directory. Renaming that one seems to have allowed Mylar to start

**Comment by evilhero** [comment]
There is no ```bs4``` directory in the root directory. There was  a very long time ago (maybe yr+), but that was moved into the ```lib``` folder for consistency.

If you have a ```bs4``` directory in the root, it will cause problems when it goes to load things as it's an outdated version (and linking problems on top of that). All the references in Mylar will try to load the ```bs4``` module from the ```lib``` folder, and failing that (for whatever reason) it will attempt to load the OS version.

I've seen the stale ```bs4``` entries happen when people install from source usually. The source installations do not delete moved/removed directories, and sometimes files as well. Source installs would require a complete removal / install to keep things synchronous with git. The updater within Mylar will update from source, but it also suffers from these stale entries happening on occasion. I don't believe this occurs with git clone installations as that keeps things synchronous with git.

**Comment by Serneum** [comment]
My installation is pulled from Git. The updater typically fails because I have some cfg files:
```
	new file:   post-processing/sabnzbd/autoProcessComics.cfg
	new file:   post-processing/sabnzbd/autoProcessComics.py
```

To update, I stash those, pull the latest, unstash that file, and start the server. My guess is that I could probably put those files elsewhere. If the database of data/config is stored elsewhere, I can do a complete removal and a new clone. Also, should I open a new issue since now this is tangential to the original issue?

**Comment by evilhero** [comment]
both of those files don't exist in those locations. They should only exist in: 
```post-processing/autoProcessComics.py post-processing/autoProcessComics.cfg```

**Comment by Serneum** [comment]
Yep, sorry, I put them in that dir and pointed Sab at it. I removed it from my original comment as I was able to find a better way. The main point was mostly just that Git didn't remove `bs4` when I would manually update

**Comment by evilhero** [comment]
Yeah I redid the updater so it would take into account deleted/stale items and basically stash those into a new subdirectory (in case changes wanted to be kept or there was a problem), but I have not pushed it out mainly due to problems with Windows-based machines. This was using source installation files though, the git stuff there has to be a way for it to forcibly remove stale entries somehow - or figure out why it's not in some cases. 

We're typing minutes apart here lol...
